gammapy/image/core.py

```diff
@@ -1088,3 +1091,82 @@ def assert_allclose(image1, image2, check_wcs=True):
             assert_wcs_allclose(image1.wcs, image2.wcs)
         else:
             raise ValueError('One image has `wcs==None` and the other does not.')
+
+    def convolve(self, kernel, fft=False, mode='reflect', **kwargs):
```

Comment:
For `SkyImage.convolve` my suggestions would be:

Remove the `fft` and `mode` options. Remove the padding and cropping stuff. Keep the implementation and tests simple. Mention that `kwargs` are passed to [scipy.ndimage.convolve](http://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.convolve.html) in the docstring, instead of exposing on arg from there, `mode`, but not the others.

`astropy.convolution` has a ton of code and time spent on trying to support both FFT and normal convolution, and there's several open issues for the FFT version for years. I strongly feel that we don't want to go down that route and spend our time on that, we really don't need FFT convolve because the only cases where convolution speed matters is large images with small kernels.